test/Transforms/memory_banking_multi_dim.mlir

```diff
@@ -1,4 +1,5 @@
 // RUN: circt-opt %s -split-input-file -memory-banking="banking-factor=2 dimension=1" | FileCheck %s --check-prefix RANK2-BANKDIM1
+// RUN: circt-opt %s -split-input-file -memory-banking="banking-factor=2" | FileCheck %s --check-prefix GETGLOBAL
```

Comment:
You may have handled this in an earlier PR, but what about corner cases, e.g., `banking-factor=0`, `dim_size % banking-factor != 0`, etc.?

Reply:
Yes, in these cases, it will early signal failure: https://github.com/llvm/circt/blob/4a062a2a1cda5052f61c700aeb204ac4ba7d2f55/lib/Transforms/MemoryBanking.cpp#L381
or raise an assertion error: https://github.com/llvm/circt/blob/4a062a2a1cda5052f61c700aeb204ac4ba7d2f55/lib/Transforms/MemoryBanking.cpp#L80

Reply:
Awesome!